lib/tasks/grade.rake

```diff
@@ -1,4 +1,223 @@
+# POTENTIAL IMPROVEMENTS
+# 1. Store personal token in ~/firstdraft.yml or Windows/Nitrous equivalent
+# 2. Task to update tests (perhaps automatically if checksum doesn't match)
+# 3. Associate pull request / commit to each submission
+# 4. Make 'descriptive' option more efficient
+
+# TESTING NEEDED
+# A. SSL & 'open' & 'rspec' commands on Windows, Nitrous
+
 desc "Grade project"
-task grade: :environment do
-  puts `bundle exec rspec --order default --format documentation --format j --out test_output.json --format html --out test_output.html`
+task :grade do # if needed in the future, add => :environment
+
+  options = {}
+  OptionParser.new do |opts|
+    opts.on("-v", "--verbose", "Show detailed test results") do |v|
+      options[:verbose] = v
+    end
+  end.parse!
+
+  begin
+    config_file_name_base = ".firstdraft_project.yml"
+    config_file_name = Rails.root.join(config_file_name_base)
+    config = YAML.load_file(config_file_name)
+    project_token = config["project_token"]
+    submission_url = config["submission_url"]
+  rescue
+    abort("ERROR: Does the file .firstdraft.yml exist?")
```

Comment:
In general, we should try to avoid `begin`/`rescue`/`end` unless we really need it to handle unforeseen errors (e.g. handling unreliable external API data).

In this case, could we do a conditional with `File.file?(Rails.root.join(".firstdraft_project.yml"))`?